bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/enums/ContentType.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hue.internal.api.dto.clip2.enums;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Enum for content type of Resource instances
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public enum ContentType {
+    ADD, // resource being added; contains (maybe) all fields
```

Comment:
Consider:
```suggestion
    @SerializedName("add")
    ADD, // resource being added; contains (maybe) all fields
```

etc.

That way you shouldn't need `toUpperCase()` below, and it can be directly serialized/deserialized.

Reply:
It is an interesting suggestion. In fact I was not aware that GSON could directly serialize and deserialize enums in that way. However as mentioned above, there are 15 other enums that are already coded using my own style, so I think it would be confusing to switch horses for the 16th such enum. We should either go for all or nothing. => WDYT?

Reply:
PS also my current coding style provides fall through results in the (albeit unexpected) cases that the deserialized DTO may have a null or unknown value; so it does provide a more fail safe approach than does your proposal.

Reply:
> PS also my current coding style provides fall through results in the (albeit unexpected) cases that the deserialized DTO may have a null or unknown value; so it does provide a more fail safe approach than does your proposal.

I think that would require a custom deserializer, but I haven't verified it.

> We should either go for all or nothing. => WDYT?

I see nothing wrong in an incremental approach, so the enum introduced in this PR _could_ utilize this approach (if desired). For sure refactoring of other enums should be done in other PR's.

Reply:
Ok

Reply:
> I think that would require a custom deserializer, but I haven't verified it.

I mean, not required, but otherwise I suspect a `JsonSyntaxException` will be thrown. When parsing a larger document where other parts are still considered valid, this may not be desirable. However, when parsing a single event, this might be acceptable, since I guess the event is of no use if the `type` is unknown (?), so the error could just be handled.

Reply:
If GSON encounters and unexpected enum value, I am not sure if it throws an exception or sets the enum to null. I did not test it, but googling seems to suggest the latter.

But anyway I decided not to go down the path of a custom deserializer, and (for the existing enums) I will stick with the old String method (which is in fact already a kind of custom deserializer).

However for the Event.contentType I decided to follow your suggestion (see latest commit). And if in future any user does encounter an event without a contentType, or with an invalid one, then we can handle that in a separate PR.